ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java

```diff
@@ -319,18 +319,27 @@ private synchronized void checkTtdEta(final PowBlock currentHead) {
 
     final long now = timeProvider.getTimeInSeconds().longValue();
 
-    final long timeDiff = now - latestBlock.getBlockTimestamp().longValue();
-
-    final long secondsToTTD =
+    final UInt256 secondsToTTD =
         specConfigBellatrix
             .getTerminalTotalDifficulty()
             .subtract(latestBlock.getTotalDifficulty())
-            .divide(averageTdPerSeconds)
-            .subtract(timeDiff)
-            .toLong();
+            .divide(averageTdPerSeconds);
+
+    // avoid negative time diff due to time discrepancy with EL
+    final UInt256 timeDiff =
+        UInt256.valueOf(Math.max(0, now - latestBlock.getBlockTimestamp().longValue()));
+
+    final long secondsToTTDAdjusted;
+    // by subtracting timeDiff we could go after the merge and get negative result,
+    // let's avoid that.
+    if (secondsToTTD.greaterThan(timeDiff)) {
+      secondsToTTDAdjusted = secondsToTTD.subtract(timeDiff).toLong();
+    } else {
+      secondsToTTDAdjusted = secondsToTTD.toLong();
+    }
```

Comment:
i wonder if we basically catch the arithmetic exception and just put to debug if it does happen, and skip the TTD line if it fails...

Reply:
Yeah.. let's see if happen again. It should not IMO... If yes i'll do that